### PR TITLE
Add arm64 builds

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR adds targets for `darwin-arm64` and `linux-arm64` to the generated Makefile for `--language go`.

Other changes

- Bump the used `architect` version in generated workflows
- Add `darwin-arm64` and `linux-arm64` targets to build matrix in workflows

Disclaimer: I have no idea how Makefiles work, I just copied around stuff until it worked on my machine:tm:

I would say, treat this PR with low priority..

### CODE REVIEW

**Review by MarcelMue — APPROVED** [review]

**Comment by ubergesundheit** [comment]
Do you have any advice on how to go forward testing this without too much disruption? Make a beta release of devctl to use with a branch of giantswarm/github?

**Comment by ubergesundheit** [comment]
Ping @sig-ux just for awareness

I'm planning to revisit this PR in the near future

**Comment by kopiczko** [comment]
@ubergesundheit what I usually do is `make install` and then `devctl gen` against my test repository. It's here if you need inspiration: https://github.com/kopiczko/test-gh-workflows.

**Review by kopiczko — APPROVED** [review]

File: cmd/gen/workflows/flag.go [hunk 1]
```diff
@@ -13,22 +13,29 @@ import (
...
+	cmd.Flags().VarP(gen.NewArchitectureSliceFlagValue(&f.Architectures, gen.ArchitectureSlice{gen.ArchitectureDarwin, gen.ArchitectureLinux}), flagArchitecture, "a", fmt.Sprintf(`The architectures to build release artifacts for through workflows. Possible values: <%s>`, strings.Join(gen.AllArchitectures(), "|")))
 }
 
 func (f *flag) Validate() error {
 	if len(f.Flavours) == 0 {
 		return microerror.Maskf(invalidFlagError, "--%s must be one of: %s", flagFlavour, strings.Join(gen.AllFlavours(), ", "))
 	}
 
+	if len(f.Architectures) == 0 {
+		return microerror.Maskf(invalidFlagError, "--%s must contain only values of: %s", flagArchitecture, strings.Join(gen.AllArchitectures(), ", "))
```
Comment (kopiczko) [inline cmd/gen/workflows/flag.go:37]: Why not all of them always?

**Comment by ubergesundheit** [comment]
Tested in https://github.com/ubergesundheit/test-gh-workflows/actions/runs/667938810 which created binaries for https://github.com/ubergesundheit/test-gh-workflows/releases/tag/v15.0.4

File: cmd/gen/workflows/flag.go (hunk 1 above)
Comment (ubergesundheit) [reply cmd/gen/workflows/flag.go:37]: I changed my mind. 2754453 enables all architectures by default :)

Comment (kopiczko) [reply cmd/gen/workflows/flag.go:37]: I don't see why someone would care to disable any of the architectures. This is only for the binaries attached to the release..

Comment (kopiczko) [reply cmd/gen/workflows/flag.go:37]: Just came to mind: so far we were _always_ releasing for darwin and linux. If we _always_ release for darwin then we should _always_ release both amd64 and arm64. Then we can get rid of the flag and the enum and simplify the code here.

Comment (ubergesundheit) [reply cmd/gen/workflows/flag.go:37]: I’ll create another PR. 👍

**Pull request merged**